tsconfig.json

```diff
@@ -11,7 +11,8 @@
     "noImplicitThis": true,
     "noUnusedLocals": true,
     "skipLibCheck": true,
-    "strict": true
+    "strict": true,
+    "lib": ["ES2017"]
```

Comment:
do you want to use `ES2017`, or use a newer version (or `ESNext`)?

Reply:
Setting it to something like ESNext would be problematic as it would mean that features not-yet-supported by some browsers could make it into the app (which are not polyfilled or otherwise transpiled).

ES2017 was offered by an error message, which happens to be the lowest version that supports all the features we currently use and is well-supported by all browsers we'd expect to see:
https://caniuse.com/async-functions,object-values,object-entries,mdn-javascript_builtins_object_getownpropertydescriptors,pad-start-end,mdn-javascript_grammar_trailing_commas_trailing_commas_in_functions

This config in the type checker essentially works as our "gate," so only the code and features able to run in production will be allowed.

We can always add individual features too, but those discussions will be as needed:
https://www.typescriptlang.org/tsconfig#individual-library-components

---

Note: Adding it here was necessary to _increase_ the list of allowed features rather than to restrict them. The effective default was lower than ES2017, and it refused to build.